ChainBridge documentation (#24)
0xPolygon/polygon-sdk-docs: docs/chainbridge/setup/deploy-contracts.md

---
id: deploy-contracts
title: Deploy Contracts
---

In this section, you will deploy required contracts to Polygon PoS and Polygon SDK chain by `cb-sol-cli`. 

> To use `cb-sol-cli`, you need to clone the codes from repository and build them.

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

> To use `cb-sol-cli`, you need to clone the codes from repository and build them.

Reviewer: > In this section, you will deploy required contracts to Polygon PoS and Polygon SDK chain by `cb-sol-cli`. 

☝️ this line can stay

> To use `cb-sol-cli`, you need to clone the codes from repository and build them.

Reviewer: cc @Kourin1996

> To use `cb-sol-cli`, you need to clone the codes from repository and build them.

Author: Reverted at c687dcd703d6f88a5ea490344b55128e695db883

```bash
# Setup for cb-sol-cli command
$ git clone https://github.com/ChainSafe/chainbridge-deploy.git
$ cd chainbridge-deploy/cb-sol-cli
$ make install
```

> Firstly, we will deploy contracts to Polygon PoS chain by `cb-sol-cli deploy` command. `--all` flag make the command deploy all contracts including Bridge, ERC20 Handler, ERC721 Handler, Generic Handler, ERC20, and ERC721 contract. In addition, it'll set the default relayer account address and the threshold.

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

```bash
# Deploy all required contracts into Polygon PoS chain
$ cb-sol-cli deploy --all --chainId 99 \
  --url https://rpc-mumbai.matic.today \
  --privateKey [ADMIN_ACCOUNT_PRIVATE_KEY] \
  --relayers [RELAYER_ACCOUNT_ADDRESS] \
  --relayerThreshold 1
```

:::info ChainID

> You'll need to specify `chainId` to deploy Bridge contract. It's the uint8 value in order to identify chains in Bridge contract. You don't have to give the same chain ID used in chain. Give `99` in `chainId` here because the chain ID of Mumbai testnet is `80001` that can't be represented uint8.

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

:::

:::info JSON-RPC URL for Polygon PoS

> `https://rpc-mumbai.matic.today` is one of the JSON-RPC URLs Polygon official provides, which may have traffic or rate-limits. I would advise you to obtain your JSON-RPC URL by an external service like `Infura` because deploying contracts will send many queries.

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

:::

:::caution

> The default gas price in `cb-sol-cli` is `20000000` (`0.02 Gwei`). To set appropriate gas price in transaction, please give the value in `--gasPrice`.

Reviewer: ```suggestion
The default gas price in `cb-sol-cli` is `20000000` (`0.02 Gwei`). To set appropriate gas price in transaction, please set the value using the `--gasPrice` argument.
```

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

Gas Limit:   8000000
Gas Price:   20000000
Deploy Cost: 0.00029065308

Options
=======
Chain Id:    <CHAIN_ID>
Threshold:   <RELAYER_THRESHOLD>
Relayers:    <RELAYER_ACCOUNT_ADDRESS>
Bridge Fee:  0
Expiry:      100

Contract Addresses
================================================================
Bridge:             <BRIDGE_CONTRACT_ADDRESS>
----------------------------------------------------------------
Erc20 Handler:      <ERC20_HANDLER_CONTRACT_ADDRESS>
----------------------------------------------------------------
Erc721 Handler:     <ERC721_HANDLER_CONTRACT_ADDRESS>
----------------------------------------------------------------
Generic Handler:    <GENERIC_HANDLER_CONTRACT_ADDRESS>
----------------------------------------------------------------
Erc20:              <ERC20_CONTRACT_ADDRESS>
----------------------------------------------------------------
Erc721:             <ERC721_CONTRACT_ADDRESS>
----------------------------------------------------------------
Centrifuge Asset:   Not Deployed
----------------------------------------------------------------
WETC:               Not Deployed
================================================================
```

> Also, you will deploy contracts to Polygon SDK chain.

Reviewer: ```suggestion
Now we may deploy the contracts to the Polygon SDK chain.
```

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

```bash
# Deploy all required contracts into Polygon SDK chain
$ cb-sol-cli deploy --all --chainId 100 \
  --url http://localhost:10002 \
  --privateKey [ADMIN_ACCOUNT_PRIVATE_KEY] \
  --relayers [RELAYER_ACCOUNT_ADDRESS] \
  --relayerThreshold 1
```

> You need the contract addresses in the next section.

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e
